### ISSUE TITLE
Option to facet by date using month or year

### ISSUE THREAD

**Issue by simonw** [simonw/datasette#483]
Facet by date (from #481) can take datetimes and facet them by the day component.

https://v0-28.datasette.io/fixtures/facetable?_facet_date=created

I'd like to also be able to facet by month or year.

I'm not sure what the best way to achieve this is. Could be two more Facet classes (YearFacet and MonthFacet) but I think it might be nicer if the existing DateFacet could take an optional argument that changed its behaviour. But... if I do that, do I expose it in the UI somewhere or is it only available to URL-hackers?

**Comment by simonw** [comment]
Here's a UI concept:

<img width="418" alt="fixtures__facetable__15_rows" src="https://user-images.githubusercontent.com/9599/58219350-551be500-7cbf-11e9-84ae-3b0386ae2063.png">

**Comment by simonw** [comment]
Would this be useful for other facets? How many facet types are likely to have a small number of options that could be linked to in this way?

**Comment by jcmkk3** [comment]
I think that location information is one of the other common pieces hierarchical data. At least one that is general enough that extra dimensions could be auto-generated.

**Comment by simonw** [comment]
Yes there's definitely something exciting to be done with location facets.

The easiest one would be a radius-distance-from-a-point facet (5km, 10km etc). A more sophisticated thing might be possible on top of GeoJSON and SpatiaLite - that's probably something I would put in a plugin rather than shipping in Datasette itself.

**Comment by simonw** [comment]
I'm going to stick with `?_facet_date=` for this and use the not-yet-fully-baked JSON alternative syntax. Probably something like this:

    ?_facet_date={"column":"created","interval":"month"}
